src/main/controller/UserController.ts

```diff
@@ -119,8 +100,18 @@ export class UserController {
         const userRepository = transactionalEntityManager.getCustomRepository(
           UserRepository
         );
-        const result = await userRepository.save(user);
-        return res.status(201).send({ ...result, firebaseId: undefined });
+        const savedUser = await userRepository.save(user);
+        try {
+          await this.firebaseService.createUserFromEmailAndPassword(
+            savedUser.id,
+            userDTO.email,
+            userDTO.password
+          );
+        } catch (e) {
+          await userRepository.delete(savedUser);
```

Comment:
Would it be better to register in firebase first then DB, and then delete from firebase if it fails (like the original FE design)? I think it's more likely that firebase will return an error from registering with an existing email.

Reply:
Oh crap sorry I did not see this.

Hmm I see, so then in our controller we would need to generate a random uid, pass it to firebase, and make sure the user saved has that same uid.

I think I like the idea of deleting our user in DB upon failure rather than deleting our user in firebase. Deleting user in DB should be much faster than deleting user in firebase, although I do agree that registering with an existing email would probably have more errors.

Reply:
Sure yeah either works!